### PR TITLE
Doc: Add monitoring for serverless

### PR DESCRIPTION
**Pull request by karenzone**
Closes: #15622

**PREVIEW:** 
* https://logstash_15636.docs-preview.app.elstc.co/guide/en/logstash/master/monitoring-with-ea.html
* https://logstash_15636.docs-preview.app.elstc.co/guide/en/logstash/master/serverless-monitoring-with-elastic-agent.html

### CODE REVIEW

**Comment by karenzone** [comment]
Adds monitoring info for serverless so that it is easy to find and provides a direct path to the info that serverless users need.

Screen Shot 2023-11-29 at 10 23 18 AM

File: docs/static/monitoring/monitoring-ea-intro.asciidoc [hunk 1]
```diff
@@ -13,13 +13,15 @@ integrations for new data sources, security protections, and more.
 
 You can use {agent} to collect monitoring data for:
 
+* Serverless dashboards
```
Comment (karenzone) [inline docs/static/monitoring/monitoring-ea-intro.asciidoc:16]: @karenzone Note to self:  Placeholder to be fleshed out with links and descriptive text

**Comment by karenzone** [comment]
@robbavey Here's what I have. So far, I've focused on the structure, with the understanding that implementation details will be different for serverless.

File: docs/static/monitoring/monitoring-ea-serverless.asciidoc [hunk 2]
```diff
@@ -0,0 +1,125 @@
+[[serverless-monitoring-with-elastic-agent]]
+=== Collect {ls} monitoring data for {serverless-full} dashboards
```
Comment (robbavey) [inline docs/static/monitoring/monitoring-ea-serverless.asciidoc:2]: Do we have a standard way to express serverless environments? I wonder if we could express it differently, something like:

Collect Logstash monitoring data for dashboards (Serverless)
Collect Logstash monitoring data for dashboards in a serverless environment

File: docs/static/monitoring/monitoring-ea-serverless.asciidoc [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+* <<add-logstash-integration-ead>>
+* <<add-agent-to-fleet-ead>>
+* <<view-assets-ead>>
+* <<view-data-dashboard>>
+
+Check out {fleet-guide}/elastic-agent-installation.html[Installing {agent}] in the _Fleet and Elastic Agent Guide_ for more info.
+
+[discrete]
+[[add-logstash-integration-svrless]]
+==== Add the {agent} {ls} integration to monitor host logs and metrics
```
Comment (robbavey) [inline docs/static/monitoring/monitoring-ea-serverless.asciidoc:54]: We'll need to write new content here for serverless

**Review by robbavey — COMMENTED** [review]
I like the general shape, and this is looking good. 

We will need to create the equivalent content to find integrations with serverless, and ensure that users create an observability project ahead of time

File: docs/static/monitoring/monitoring-ea-serverless.asciidoc [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+====
+include::monitoring-prereq-disable-default.asciidoc[]
+====
+
+[[define-cluster__uuid-svrless]]
+.Specify the target cluster_uuid (optional)
+[%collapsible]
+====
+include::monitoring-prereq-define-cluster.asciidoc[]
+====
```
Comment (karenzone) [inline docs/static/monitoring/monitoring-ea-serverless.asciidoc:27]: @karenzone: I believe this one goes away. VALIDATE!

File: docs/static/monitoring/monitoring-ea-serverless.asciidoc [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+[[create-user-svrless]]
+.Create a monitoring user (standalone agent only)
+[%collapsible]
+====
+include::monitoring-prereq-create-user.asciidoc[]
+====
```
Comment (karenzone) [inline docs/static/monitoring/monitoring-ea-serverless.asciidoc:34]: @karenzone: I believe this one goes away, too.  VALIDATE!

File: docs/static/monitoring/monitoring-ea-intro.asciidoc [hunk 6]
```diff
@@ -11,18 +11,21 @@ Plus you have the option to manage all of your monitoring agents from a central
 types of data to a host. Each agent has a single policy you can update to add
 integrations for new data sources, security protections, and more.
 
-You can use {agent} to collect monitoring data for:
+You can use {agent} to collect {ls} monitoring data on:
 
-* <<dashboard-monitoring-with-elastic-agent,Dashboards>>. {agent} collects
+* <<serverless-monitoring-with-elastic-agent,{serverless-full} dashboards>>. {agent} collects
```
Comment (robbavey) [inline docs/static/monitoring/monitoring-ea-intro.asciidoc:16]: WDYT about putting Elastic cloud/self-managed first, at least until serverless is GA?

**Review by robbavey — APPROVED** [review]
This is great!

I have one question about the order of serverless vs cloud/self-managed in the text for the intro - whether we should put serverless lower in the text, at least until it is GA

File: docs/static/monitoring/monitoring-ea-serverless.asciidoc (hunk 4 above)
Comment (karenzone) [reply docs/static/monitoring/monitoring-ea-serverless.asciidoc:27]: CONFIRMED

File: docs/static/monitoring/monitoring-ea-serverless.asciidoc (hunk 5 above)
Comment (karenzone) [reply docs/static/monitoring/monitoring-ea-serverless.asciidoc:34]: CONFIRMED

**Review by robbavey — APPROVED** [review]
LGTM

**Comment by karenzone** [comment]
51383

**Comment by karenzone** [comment]
@logstashmachine backport 8.12

**Comment by karenzone** [comment]
@logstashmachine backport 8.11

**Pull request merged**